src/media/js/addon/actions/addon.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+import {createAction} from 'redux-actions';
+import req from 'request';
+import Url from 'urlgray';
+import urlJoin from 'url-join';
+
+
+export const FETCH_OK = 'ADDON_ADDON__FETCH_OK';
+const fetchOk = createAction(FETCH_OK);
+
+export const FETCH_VERSIONS_OK = 'ADDON_ADDON__FETCH_VERSIONS_OK';
+const fetchVersionsOk = createAction(FETCH_VERSIONS_OK);
+
+
+export function fetch(addonSlug) {
+  /*
+    Fetch add-on.
+  */
+  return (dispatch, getState) => {
+    const apiArgs = getState().apiArgs || {};
+    const addonUrl = Url(
+      urlJoin(process.env.MKT_API_ROOT, 'extensions/extension/', addonSlug)
+    ).q(apiArgs);
+
+    if (process.env.NODE_ENV === 'test') {
```

Comment:
Network level seems like the right thing to me but I guess it doesn't really matter. Either way you can just stub `req` in the test code, right? Why put code like this in the production file?

```js
function testFetchAddon() {
    sinon
        .stub(req, 'get')
        .withArgs('/api/addon/25')
        .returns(new Promise(function(resolve) {
            resolve(addonFactory());
        });
    assertThingDoesWhatItDoes();
}
```

If you can mock at the network level then you can tell the mocking library to return `addonFactory()` and `GET /api/addon/25` which seems way cleaner. The only thing I could find was `sepia` and it doesn't look super well maintained. I also don't know if you're running these tests in Node or in a browser. You could use `sinon` in the browser like we do on other projects.